examples/geometry_optimization.jl

```diff
@@ -7,12 +7,14 @@ using DFTK
 using Optim
 using LinearAlgebra
 using Printf
+using Unitful
+using UnitfulAtomic
 
 kgrid = [1, 1, 1]       # k-Point grid
-Ecut = 5                # kinetic energy cutoff in Hartree
+Ecut = 5u"hartree"      # kinetic energy cutoff
 tol = 1e-8              # tolerance for the optimization routine
-a = 10                  # lattice constant in Bohr
-lattice = a * Diagonal(ones(3))
+a = 10 * u"bohr"        # lattice constant
+lattice = austrip(a) * Diagonal(ones(3))
```

Comment:
Do you need to austrip here? I think this should work with the units too shouldn't it?

Reply:
Indeed we don't really want the user to know about austrip